drivers/rtc/rtc_ll_stm32.c

```diff
@@ -76,46 +131,179 @@ struct rtc_stm32_config {
 #endif
 };
 
+#ifdef CONFIG_RTC_ALARM
+struct rtc_stm32_alrm {
+	LL_RTC_AlarmTypeDef ll_rtc_alrm;
+	/* supported masks, values from RTC_ALARM_TIME_MASK */
+	uint16_t supported_mask;
+	/* user-defined mask, values from RTC_ALARM_TIME_MASK */
+	uint16_t user_mask;
+	rtc_alarm_callback user_callback;
+	void *user_data;
+	bool is_pending;
+};
+#endif /* CONFIG_RTC_ALARM */
+
 struct rtc_stm32_data {
 	struct k_mutex lock;
+#ifdef CONFIG_RTC_ALARM
+	struct rtc_stm32_alrm rtc_alrm_a;
+	struct rtc_stm32_alrm rtc_alrm_b;
+#endif /* CONFIG_RTC_ALARM */
 };
 
-static int rtc_stm32_configure(const struct device *dev)
+#ifdef CONFIG_RTC_ALARM
+static inline ErrorStatus ll_func_init_alarm(RTC_TypeDef *rtc, uint32_t format,
+					LL_RTC_AlarmTypeDef *ll_alarm_struct, uint16_t id)
 {
-	const struct rtc_stm32_config *cfg = dev->config;
+	/* RTC write protection is disabled & enabled again inside LL_RTC_ALMx_Init functions */
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	return LL_RTC_ALARM_Init(rtc, format, ll_alarm_struct);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		ll_alarm_struct->AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_DATE;
+		return LL_RTC_ALMA_Init(rtc, format, ll_alarm_struct);
+	}
+	if (id == RTC_STM32_ALRM_B) {
+		ll_alarm_struct->AlarmDateWeekDaySel = LL_RTC_ALMB_DATEWEEKDAYSEL_DATE;
+		return LL_RTC_ALMB_Init(rtc, format, ll_alarm_struct);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+	return 0;
+}
 
-	int err = 0;
+static inline void ll_func_clear_alarm_flag(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
 
-	uint32_t hour_format     = LL_RTC_GetHourFormat(RTC);
-	uint32_t sync_prescaler  = LL_RTC_GetSynchPrescaler(RTC);
-	uint32_t async_prescaler = LL_RTC_GetAsynchPrescaler(RTC);
+	LL_RTC_ClearFlag_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_ClearFlag_ALRA(rtc);
+	} else {
+		LL_RTC_ClearFlag_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
 
-	LL_RTC_DisableWriteProtection(RTC);
+static inline uint32_t ll_func_is_active_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
+
+	return LL_RTC_IsActiveFlag_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		return LL_RTC_IsActiveFlag_ALRA(rtc);
+	}
+	if (id == RTC_STM32_ALRM_B) {
+		return LL_RTC_IsActiveFlag_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+	return 0;
+}
+
+static inline void ll_func_enable_interrupt_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
+
+	LL_RTC_EnableIT_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_EnableIT_ALRA(rtc);
+	} else {
+		LL_RTC_EnableIT_ALRB(rtc);
```

Comment:
Did you try to compile the alarm subsystem on series with only one alarm other than F1 (like C0 or F0) ? I don't think it works since all functions and variables with ALRB aren't defined.

Reply:
There is a `CONFIG_SOC_SERIES_STM32F1X` just before.

Reply:
Good catch !!

Reply:
Fixed.